torch_dftd/nn/base_dftd_module.py

```diff
@@ -111,22 +111,15 @@ def calc_energy_and_forces(
                 "forces": (n_atoms, 3)
                 "stress": (6,)
         """
+        pos.requires_grad_(True)
         if cell is not None:
             # pos is depending on `cell` size
             # We need to explicitly include this dependency to calculate cell gradient
             # for stress computation.
             # pos is assumed to be inside "cell", so relative position `rel_pos` lies between 0~1.
-            if batch is None:
-                rel_pos = torch.mm(pos, torch.inverse(cell))
-                pos = torch.mm(rel_pos.detach(), cell)
-            else:
-                # pos (n_atoms, 3) & cell (bs, 3, 3)
-                rel_pos = torch.bmm(pos[:, None, :], torch.inverse(cell)[batch])[:, 0]
-                # pos (n_atoms, 1, 3) * cell (n_atoms, 3, 3) -> (n_atoms, 3)
-                pos = torch.bmm(rel_pos[:, None, :].detach(), cell[batch])[:, 0]
+            assert isinstance(shift, Tensor)
+            shift.requires_grad_(True)
```

Comment:
`pos` and `shift` is explicitly defined as gradient variables here.
The effect on `cell` should both applied to `pos` and `shift`. In the previous implementation, it is calculated by the redefenition of `pos` and making call graph.
In the new implementation, both effect is calculated explicitly later and thus `pos` no longer need to have calculation graph from `cell`. This change removes `cell[batch]` calculation.